kii_thing_if.c

```diff
@@ -277,6 +278,30 @@ static prv_bool_t prv_set_firmware_version_resource_path(
   return TRUE;
 }
 
+static prv_bool_t prv_set_thing_type_resource_path(
+        kii_t* kii,
+        char* resource_path,
+        size_t resource_path_len)
+{
+    if (resource_path_len <=
+            CONST_STRLEN(THING_IF_APP_PATH) +
```

Comment:
@satoshikumano 

The reason I adopt this style is to reduce stack size and binary size. Using local variable and/or functions are increase the stack size and binary size.

So what do you think about using macro like following:

```c
if (resouce_path_len <=
        THING_TYPE_PATH_LEN(
            kii->kii_core.app_id,
            kii->kii_core.author.author_id)) {
```